### PR TITLE
(Needs testing) Makes all doors not process

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Removes all doors from processing , cutting down around 1500~ machinery from 10000 machinery processing roundstart
Very likely will break in edge cases i can't test or even think of

## Why It's Good For The Game
Holy fuck why do we still process every fucking door this is 2021
## Changelog
:cl:
fix: Fixed doors processing
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -310,17 +310,7 @@ GLOBAL_LIST_EMPTY(wedge_icon_cache)
...
-		regainMainPower()
-
-	if(backup_power_lost_until > 0 && world.time >= backup_power_lost_until)
-		regainBackupPower()
-
-	else if(electrified_until > 0 && world.time >= electrified_until)
-		electrify(0)
-
-	..()
+	return PROCESS_KILL
```
Comment (drexample) [inline code/game/machinery/doors/airlock.dm:313]: Why

Comment (MLGTASTICa) [reply code/game/machinery/doors/airlock.dm:313]: So it doesn't process?

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -488,18 +478,21 @@ There are 9 wires.
 	return src.isWireCut(AIRLOCK_WIRE_BACKUP_POWER1) || src.isWireCut(AIRLOCK_WIRE_BACKUP_POWER2)
 
 /obj/machinery/door/airlock/proc/loseMainPower()
-	main_power_lost_until = mainPowerCablesCut() ? -1 : world.time + SecondsToTicks(60)
+	main_power_lost_until = mainPowerCablesCut() ? -1 : SecondsToTicks(60)
+	spawn(main_power_lost_until) regainMainPower()
```
Comment (Doster-d) [inline code/game/machinery/doors/airlock.dm:482]: Maybe you better to use this?
```suggestion
	addtimer(CALLBACK(src, .proc/regainMainPower), main_power_lost_until)
```

Comment (MLGTASTICa) [reply code/game/machinery/doors/airlock.dm:482]: spawn is less resource intensive than timers , and i don't need the precision of one here

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -488,18 +478,23 @@ There are 9 wires.
...
-	main_power_lost_until = mainPowerCablesCut() ? -1 : world.time + SecondsToTicks(60)
+	main_power_lost_until = mainPowerCablesCut() ? -1 : SecondsToTicks(60)
+	if(main_power_lost_until > 0)
+		spawn(main_power_lost_until) regainMainPower()
 
 	// If backup power is permanently disabled then activate in 10 seconds if possible, otherwise it's already enabled or a timer is already running
 	if(backup_power_lost_until == -1 && !backupPowerCablesCut())
-		backup_power_lost_until = world.time + SecondsToTicks(10)
+		backup_power_lost_until = SecondsToTicks(10)
+		spawn(backup_power_lost_until) regainBackupPower()
```
Comment (AlexMorgan3817) [inline code/game/machinery/doors/airlock.dm:488]: Use addtimer

File: code/game/machinery/doors/airlock.dm [hunk 4]
```diff
@@ -488,18 +478,23 @@ There are 9 wires.
 	return src.isWireCut(AIRLOCK_WIRE_BACKUP_POWER1) || src.isWireCut(AIRLOCK_WIRE_BACKUP_POWER2)
 
 /obj/machinery/door/airlock/proc/loseMainPower()
-	main_power_lost_until = mainPowerCablesCut() ? -1 : world.time + SecondsToTicks(60)
+	main_power_lost_until = mainPowerCablesCut() ? -1 : SecondsToTicks(60)
+	if(main_power_lost_until > 0)
+		spawn(main_power_lost_until) regainMainPower()
```
Comment (AlexMorgan3817) [inline code/game/machinery/doors/airlock.dm:483]: https://github.com/discordia-space/CEV-Eris/pull/6527#discussion_r710784776
And we don't use linear blocks like if() action(), for() action(), spawn() action()

File: code/game/machinery/doors/airlock.dm [hunk 5]
```diff
@@ -488,18 +478,23 @@ There are 9 wires.
...
 
 	// Disable electricity if required
 	if(electrified_until && isAllPowerLoss())
 		electrify(0)
 
 /obj/machinery/door/airlock/proc/loseBackupPower()
-	backup_power_lost_until = backupPowerCablesCut() ? -1 : world.time + SecondsToTicks(60)
+	backup_power_lost_until = backupPowerCablesCut() ? -1 : SecondsToTicks(60)
+	if(backup_power_lost_until > 0)
+		spawn(backup_power_lost_until) regainBackupPower()
```
Comment (AlexMorgan3817) [inline code/game/machinery/doors/airlock.dm:497]: https://github.com/discordia-space/CEV-Eris/pull/6527#discussion_r710785239

File: code/game/machinery/doors/airlock.dm [hunk 6]
```diff
@@ -535,7 +530,8 @@ There are 9 wires.
 		else
 			shockedby += text("\[[time_stamp()]\] - EMP)")
 		message = "The door is now electrified [duration == -1 ? "permanently" : "for [duration] second\s"]."
-		electrified_until = duration == -1 ? -1 : world.time + SecondsToTicks(duration)
+		electrified_until = duration == -1 ? -1 : SecondsToTicks(duration)
+		spawn(electrified_until) electrify(0)
```
Comment (AlexMorgan3817) [inline code/game/machinery/doors/airlock.dm:534]: https://github.com/discordia-space/CEV-Eris/pull/6527#discussion_r710785239

File: code/game/machinery/doors/door.dm [hunk 7]
```diff
@@ -449,22 +443,17 @@
 	update_icon()
 	update_nearby_tiles()
 	operating = 0
-
 	if(autoclose)
-		close_door_at = next_close_time()
+		spawn(normalspeed ? 150 : 5) close()
```
Comment (AlexMorgan3817) [inline code/game/machinery/doors/door.dm:447]: https://github.com/discordia-space/CEV-Eris/pull/6527#discussion_r710785239

**Review by AlexMorgan3817 — CHANGES_REQUESTED** [review]
Test this, because if every spawn will be triggered somehow, it will do strange stranges

File: code/game/machinery/doors/airlock.dm (hunk 2 above)
Comment (AlexMorgan3817) [reply code/game/machinery/doors/airlock.dm]: > spawn is less resource intensive than timers , and i don't need the precision of one here

Spawns isuncotrollable, better to use timers

Comment (MLGTASTICa) [reply code/game/machinery/doors/airlock.dm]: I don't need it to be controllable , using timers here is redundant and i don't need their functionality here at all , but whatever.

**Review by drexample — APPROVED** [review]
